### DISCUSSION TITLE
Duckdb accessing csv, excel and parquet files in onedrive

### DISCUSSION THREAD

**Discussion by yoonghm** [duckdb/duckdb#8709] [General]
Duckdb could access files stored in s3 storage. Could this feature be expanded to access files stored in onedrive, Google drive and Dropbox?

See the example here for pandas accessing a file stored in onedrive:

https://www.cdata.com/kb/tech/onedrive-python-pandas.rst

**Comment by Mause** [comment]
Google drive and Dropbox are already available via the fsspec integration, unfortunately it looks like the one drive API library is unmaintained

**Reply by yoonghm** [reply]
The lastest API to access Microsoft services (such as OneDrive, Azure's services) is Microsoft Graph. It is subjected to change. I agree with your comment. Thanks.
